orchestra/executor.py

```diff
@@ -379,6 +383,19 @@ def _transitive_reduction(graph):
 
         return inflated_graph
 
+    @staticmethod
+    def _verify_binary_archives_exist(dependency_graph):
+        for action in dependency_graph.nodes:
+            if not isinstance(action, InstallAction):
+                continue
+            if not action.binary_archive_exists() and not action.allow_build:
+                binary_archive_filename = action.build.binary_archive_filename
+                qualified_name = action.build.qualified_name
+                raise Exception(
+                    f"""Binary archive {binary_archive_filename} for {qualified_name} not found.
+                    Try `orc update` or run `orc install` with `-b`."""
+                )
```

Comment:
Very often people report "no binary archive found". You're adding the name of the archive we were looking for, which is great. What do you think about including the list of all binary archives available in the target directory?
Not sure it helps. Just thinking out loud.

Reply:
I'm not sure that reporting a list of binary archives helps either.
I don't think a message like this would be helpful to a user
> Binary archive aaaa_bbbb.tar.xz for <component> not found.
> Available archives:
>  - aaaa_yyyy.tar.xz
>  - wwww_zzzz.tar.xz
>  - asd_asd.tar.xz
>
> Try `orc update` or run `orc install` with `-b`.

The list of available archives gets quite long and the only additional information users could get from it is that a binary archive for the same commit was built at some point (looking at the first part of the filename).

Reply:
Agreed.